src/App.tsx

```diff
@@ -1,111 +1,321 @@
               </Typography>
-            } 
-            placement='top' 
-            TransitionComponent={Zoom} 
-            TransitionProps={{ timeout: 500 }}
-            enterDelay={500}
-            >
-              <Typography variant="h5" className='dataset-name'>{datasetName}</Typography>
             </Tooltip>
             <Typography variant="subtitle1">from {nodeName}</Typography>
-            <Typography variant='subtitle2'>{numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects</Typography>
-          </Grid>
-          </Grid>
-          <Grid item>
+            <Typography variant="subtitle2">
+              {numMatchingSubjects} subjects match / {datasetTotalSubjects} total subjects
+            </Typography>
+          </div>
+          <div className="col-span-2 justify-self-end">
             <ButtonGroup>
               {imageModals.sort().map((modal) => (
                 <Button key={modal} variant="text" className={modalities[modal].style}>
                   {modalities[modal].name}
                 </Button>
               ))}
             </ButtonGroup>
-          </Grid>
-        </Grid>
+          </div>
+        </div>
       </CardContent>
     </Card>
-  )
+  );
 }
 
-function ResultContainer({result} : {result: Result[]}) {
+function ResultContainer({ result }: { result: Result[] }) {
   return (
-    <Stack spacing={2}>
-          {exampleResult.map((item) =>
-          <ResultCard key={item.dataset_uuid} nodeName={item.node_name} datasetName={item.dataset_name} datasetTotalSubjects={item.dataset_total_subjects} numMatchingSubjects={item.num_matching_subjects} imageModals={item.image_modals} />
-          )}
-    </Stack>
-  )
+    <div className="grid gap-4">
+      {result.map((item) => (
+        <ResultCard
+          key={item.dataset_uuid}
+          nodeName={item.node_name}
+          datasetName={item.dataset_name}
+          datasetTotalSubjects={item.dataset_total_subjects}
+          numMatchingSubjects={item.num_matching_subjects}
+          imageModals={item.image_modals}
+        />
+      ))}
+    </div>
+  );
 }
 
-function App() {
-  const [result, setResult] = useState<Result[]>([])
+function QueryForm({
+  diagnosisOptions,
+  assessmentOptions,
+  apiQueryURL,
+  onSubmitQuery,
+}: {
+  diagnosisOptions: APIOption[];
+  assessmentOptions: APIOption[];
+  apiQueryURL: string;
+  onSubmitQuery: (url: string) => void;
+}) {
+  const [minAge, setMinAge] = useState<string | null>(null);
+  const [maxAge, setMaxAge] = useState<string | null>(null);
+  const [sex, setSex] = useState<string | null>(null);
+  const [diagnosis, setDiagnosis] = useState<string | null>(null);
+  const [isControl, setIsControl] = useState<boolean>(false);
+  const [minNumSessions, setMinNumSessions] = useState<string | null>(null);
+  const [assessmentTool, setAssessmentTool] = useState<string | null>(null);
+  const [imagingModality, setImagingModality] = useState<string | null>(null);;
```

Comment:
just a note for later: since the modalities are fixed / not fetched, we could declare the valid options in the type hint. I believe it's

`type modalities = "T1w" | "T2w"` and so on. 

Again, more for later